scripts/check-gmp.gmp

```diff
@@ -0,0 +1,1565 @@
+                )
+            elif trend in ["down", "same", "less"]:
+                end_session(im, "GMP OK: Trend is %s." % trend, NAGIOS_OK)
+            else:
+                end_session(
+                    im,
+                    "GMP UNKNOWN: Trend is unknown: %s" % trend,
+                    NAGIOS_UNKNOWN,
+                )
+        else:
+            last_report_id = task.xpath("task/last_report/report/@id")
+
+            if not last_report_id:
+                end_session(
+                    im, "GMP UNKNOWN: Report is not available", NAGIOS_UNKNOWN
+                )
+
+            last_report_id = last_report_id[0]
+            # pretty(task)
+            last_scan_end = task.xpath(
+                "task/last_report/report/scan_end/text()"
+            )
+
+            if last_scan_end:
+                last_scan_end = last_scan_end[0]
+            else:
+                last_scan_end = ""
+
+            if im.old_report(last_scan_end, params_used):
+                autofp = ""
+
+                if script_args.autofp:
+                    autofp = "autofp=%i" % script_args.autofp
+
+                # When i add the ip address in the filter by a big report with
+                # a lot vulns, then the filtering is quick. But when the
+                # report did not contain the host, then it takes a lot longer.
+                # I dont know why. I add a check for the result count.
+                # If > 500 then add the host else without host
+                #
+                #   <debug>0</debug>
+                #   <hole>413</hole>
+                #   <info>380</info>
+                #   <log>9490</log>
+                #   <warning>2332</warning>
+                #   <false_positive>842</false_positive>
+
+                result_count = task.xpath(
+                    "task/last_report/report/result_count"
+                )
+                result_count = result_count[0]
+                # sum_results = result_count.xpath(
+                #     'sum(hole/text() | warning/text() | info/text())')
+                # host = ''
+                # Host must be setted, otherwise we get results of other hosts in the reports
+                # if sum_results > 200:
+                host = script_args.hostaddress
+
+                full_report = gmp.get_reports(
+                    filter_id=last_report_id,
+                    filter="sort-reverse=id result_hosts_only=1 "
+                    "min_cvss_base= min_qod= levels=hmlgd autofp=%s "
+                    "notes=0 apply_overrides=%s overrides=%s first=1 rows=-1 "
+                    "delta_states=cgns host=%s"
+                    % (
+                        script_args.autofp,
+                        int(script_args.overrides),
+                        int(script_args.apply_overrides),
+                        host,
+                    ),
+                )
+
+                # pretty(report.xpath('report/report/filters'))
```

Comment:
Please just remove debug statements even if outcommented. They should not be included in the code.